plugins/browser-plugin-media-tracking/src/index.ts

```diff
@@ -0,0 +1,230 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import {
+  isTypeTextTrackEvent,
+  isTypeDocumentEvent,
+  boundryErrorHandling,
+  trackingOptionsParser,
+} from './helperFunctions';
+import { SnowplowMediaEvent } from './snowplowEvents';
+import { DocumentEvent, MediaEvent } from './mediaEvents';
+import { MediaEventType, TrackingOptions, RecievedTrackingOptions, EventGroup, trackedElement } from './types';
+import { BrowserPlugin, BrowserTracker, dispatchToTrackersInCollection } from '@snowplow/browser-tracker-core';
+import { buildSelfDescribingEvent, CommonEventProperties, SelfDescribingJson } from '@snowplow/tracker-core';
+import { MediaPlayerEvent } from './contexts';
+import { findMediaElem } from './findMediaElement';
+import { buildMediaEvent } from './buildMediaEvent';
+import { MediaProperty } from './mediaProperties';
+
+declare global {
+  interface HTMLVideoElement {
+    autoPictureInPicture: boolean;
+    disableRemotePlayback: boolean;
+    disablePictureInPicture: boolean;
+  }
+  interface HTMLAudioElement {
+    disableRemotePlayback: boolean;
+  }
+  interface Document {
+    pictureInPictureElement: Element;
+  }
+}
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function MediaTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+  };
+}
+
+const trackedIds: Record<string, trackedElement> = {};
+
+export function enableMediaTracking(args: { id: string; options?: RecievedTrackingOptions }) {
```

Comment:
```suggestion
export function enableMediaTracking(args: { id: string; options?: MediaTrackingOptions }) {
```

This is your public API, so I think I'd make this something like `MediaTrackingOptions`, and then the existing internal interface `TrackingOptions` into `ParsedTrackingOptions`.